1720 Eget flagg for å markere meldekort som korrigeringer

Henreich opened the pull request:
https://trello.com/c/cNYfdB7o/1720-brev-pdf-til-bruker-for-korrigert-meldekort-m%C3%A5-stemme-overrens-med-det-brukern-faktisk-ser

Henreich committed:
1720 Eget flagg for å markere meldekort som korrigeringer

Henreich committed:
1720 Slemme spotless

Henreich committed:
1720 Testfix - forventes at meldekort er markert som korrigering

hestad commented on src/main/kotlin/no/nav/tiltakspenger/meldekort/domene/Meldekort.kt:
```diff
@@ -31,6 +31,7 @@ data class Meldekort(
     val journalføringstidspunkt: LocalDateTime?,
     val varselId: VarselId? = null,
     val erVarselInaktivert: Boolean = false,
+    val korrigering: Boolean = false,
```
Synes det er litt skummelt med defaultverdi her

Henreich replied:
det luket ut en feil det, så godt poeng :D

hestad commented on src/main/kotlin/no/nav/tiltakspenger/meldekort/domene/Meldekort.kt:
```diff
@@ -70,6 +71,7 @@ data class Meldekort(
         sisteMeldeperiode: Meldeperiode,
         clock: Clock,
         brukerutfylteDager: List<MeldekortDag>,
+        korrigering: Boolean = false,
```
samme her

hestad commented on src/main/resources/db/migration/V24__meldekort_korrigering.sql:
```diff
@@ -0,0 +1,14 @@
+
+WITH meldekort_i_kjede AS (
+    SELECT id,
+           row_number() OVER (PARTITION BY meldeperiode_id ORDER BY mottatt) AS index
```
meldeperiode_id er unik per versjon av meldeperiode. du må se på meldeperiode.kjede_id

Henreich replied:
trodde jeg forsto kommentaren din, men resultatet får det til å se ut som nesten alle meldekort er korrigert 😭

Henreich replied:
oh selvfølgelig.. fordi kjede_id er jo ikke unik alene da mange meldekort har de samme periodene

Henreich committed:
1720 PR kommentarer

Henreich committed:
1720 spotless

Henreich committed:
1720 Fikser join på meldeperiode, intellij lurte meg med sitt forslag på en kolonne som ikke finnes

Henreich committed:
1720 kjede_id er ikke unik for seg selv, må derfor gruppere på sak også

fordi kjede_id er jo bare tilOgMed / fraOgMed datoer, så vil alle meldekort med samme perioder få de samme kjede_id-ene

Henreich committed:
1720 Opprettelse av korrigering-kolonnen ble flyttet til eget script for å kjøre det separat
